tests/e2e/dev.rs

```diff
@@ -0,0 +1,98 @@
+use std::env;
+use std::process::Command;
+use std::time::Duration;
+
+use assert_cmd::prelude::CommandCargoExt;
+use mime::APPLICATION_JSON;
+use reqwest::header::CONTENT_TYPE;
+use reqwest::Client;
+use rstest::*;
+use serde_json::{json, Value};
+use speculoos::assert_that;
+use tempfile::TempDir;
+use tokio::time::timeout;
+
+use crate::e2e::{
+    run_subgraphs_retail_supergraph, test_graphql_connection, GRAPHQL_TIMEOUT_DURATION,
+};
+
+const ROVER_DEV_TIMEOUT: Duration = Duration::from_secs(45);
+const ROUTER_PORT: u32 = 4123;
+
+#[fixture]
+#[once]
+fn run_rover_dev(#[from(run_subgraphs_retail_supergraph)] working_dir: &TempDir) -> String {
+    let mut cmd = Command::cargo_bin("rover").expect("Could not find necessary binary");
+    let router_url = format!("http://localhost:{}", ROUTER_PORT);
+    let client = Client::new();
+
+    cmd.args([
+        "dev",
+        "--supergraph-config",
+        "supergraph-config-dev.yaml",
+        "--router-config",
+        "router-config-dev.yaml",
+        "--supergraph-port",
+        &format!("{}", ROUTER_PORT),
+        "--elv2-license",
+        "accept",
+    ]);
+    cmd.current_dir(working_dir);
+    if let Ok(version) = env::var("APOLLO_ROVER_DEV_COMPOSITION_VERSION") {
+        cmd.env("APOLLO_ROVER_DEV_COMPOSITION_VERSION", version);
+    };
+    if let Ok(version) = env::var("APOLLO_ROVER_DEV_ROUTER_VERSION") {
+        cmd.env("APOLLO_ROVER_DEV_ROUTER_VERSION", version);
+    };
+    cmd.spawn().expect("Could not run rover dev command");
+    tokio::task::block_in_place(|| {
+        let handle = tokio::runtime::Handle::current();
+        handle.block_on(async {
+            timeout(
+                ROVER_DEV_TIMEOUT,
+                test_graphql_connection(&client, &router_url),
+            )
+            .await
+            .expect("foo")
+        })
+    })
```

Comment:
I'll admit I stole this idiom from a similar solution to the problem of `rstest` not supporting `async` `#[once]` fixtures. There might be a better way to do this and if so then I'm all ears!